### ISSUE TITLE
Doc for objects that inherit from BaseCoordinateFrame is messy

### ISSUE THREAD

**Issue by dstansby** [astropy/astropy#7458]
e.g. see this page: http://docs.sunpy.org/en/stable/api/sunpy.coordinates.frames.HeliographicStonyhurst.html#sunpy.coordinates.frames.HeliographicStonyhurst and scroll down to "Attributes summary"

Because `BaseCoordinateFrame` has a few class attributes, these get automatically documented (without any docstrings) when documenting an inheriting class. It would be nice to make these disappear somehow - maybe converting them to properties, and making the class attributes hidden by putting an underscore before their names?

**Comment by pllim** [comment]
Shouldn't this be a `sunpy` issue? They can hide whatever they want using "exclude members" directive, no?

**Comment by dstansby** [comment]
Maybe the attributes should be documented in astropy too?

**Comment by pllim** [comment]
Sure, documenting the attributes is a good idea. However, changing the code for the sake of documentation is probably not (unless there is a very good reason to do so). Thank you for bringing this to our attention!

**Comment by pllim** [comment]
And last but not the least, you are welcome to contribute!

**Comment by dstansby** [comment]
👍 to documenting the properties - if I have time I will have a look, but busy with lots of other stuff right now

**Comment by b1quint** [comment]
May I contribute by addressing this issue?

**Comment by Himanshu-Garg** [comment]
Anyone doing this?
Or should i start with it, as there is no PR mentioning this issue and last comment is 3 months old.

**Comment by b1quint** [comment]
Yeah, I was trying to help on that but I could not understand what the code was doing. I should have reported on that. If you believe you can do it, go for it!

**Comment by Himanshu-Garg** [comment]
Sure, will try

**Comment by astrojuanlu** [comment]
Still an issue:

![Screenshot_2020-10-04 HeliographicStonyhurst](https://user-images.githubusercontent.com/316517/95011076-710ef500-062e-11eb-9dc7-edc934447667.png)

**Comment by astrojuanlu** [comment]
To be more precise, some of the required info is here:

https://github.com/astropy/astropy/blob/80060446ccdd503c65a01266b92ea08a6f2fc57d/astropy/coordinates/baseframe.py#L369-L388

I suppose that writing a proper docstring for `BaseCoordinateFrame` where these properties are documented in [the NumPy style](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#google-vs-numpy) (as the rest of Astropy docstrings) should be enough, but I haven't checked.

**Comment by Shaheer-Ahmd** [comment]
Is this issue still relevant or is it resolved? Should I work on it?

**Comment by pllim** [comment]
@dstansby or @ayshih , do you know if this issue is still relevant?

**Comment by ayshih** [comment]
Docstrings for the class properties (e.g., `default_differential`) were already added in #12005.  (Amusingly, I have *zero* recollection of writing that PR...)

The class variables (`name` and `frame_attributes`) are just variables, but we could make them properties if we want to add docstrings.  Users shouldn't be modifying those anyway.

As an aside, even if the above is fixed, any frame attribute (e.g., `obstime`) currently does not have a docstring, so a way would need to be provided to set a docstring for the descriptor.

**Comment by Architrb1795** [comment]
Hi @pllim and @ayshih,

I’m considering contributing to this issue as a new Astropy contributor and wanted to check before starting any work.

From the discussion, it looks like parts of this were addressed in #12005, but some class-level attributes (e.g. `name`, `frame_attributes`) may still appear in autogenerated docs without clear docstrings.

Could you please clarify:
- Whether this issue is still considered open/relevant today?
- And if so, whether there is a preferred direction for addressing the remaining documentation concerns (e.g. documentation-only changes or code changes too)?

Happy to work on this if guidance is available. Thanks!

**Comment by ayshih** [comment]
Thanks for your interest!  In my opinion, this issue can be closed after #15259, because the remaining undocumented attributes don't really need documentation.  The documentation approach will probably have to change anyway when the frame classes are switched to be dataclasses.  I suggest you look for a different issue to work on.

**Comment by Pittu-Sharma** [comment]
Hi, I would like to work on this issue as a first contribution.
Could you please assign it to me?

**Comment by ayshih** [comment]
Sorry, this issue should be closed as completed.  Please look for other issues to work on!
